lib/config/options.rb

```diff
@@ -152,23 +121,46 @@ def respond_to_missing?(*args)
 
     protected
 
-    def descend_array(array)
-      array.map do |value|
-        if value.instance_of? Config::Options
-          value.to_hash
-        elsif value.instance_of? Array
-          descend_array(value)
-        else
-          value
-        end
+    def load_config_sources(method, source)
+      # handle yaml file paths
+      source = (Sources::YAMLSource.new(source)) if source.is_a?(String)
+      source = (Sources::HashSource.new(source)) if source.is_a?(Hash)
+
+      @config_sources ||= []
+      @config_sources.send(method, source)
+    end
+
+    def hash_deep_merge(source, dest)
+      DeepMerge.deep_merge!(
+                            source,
+                            dest,
+                            preserve_unmergeables: false,
+                            knockout_prefix:       Config.knockout_prefix,
+                            overwrite_arrays:      Config.overwrite_arrays,
+                            merge_nil_values:      Config.merge_nil_values,
+                            merge_hash_arrays:     Config.merge_hash_arrays
+                          )
+    end
+
+    def instance_of(value)
```

Comment:
`instance_of` is a strange name here, but I'm struggling coming up with a better name.  To me, `instance_of` implies you are getting another instance of something (perhaps a dup), but it's actually returning the converted value of the passed in value.  Perhaps

```suggestion
    def serialize_value(value)
```

Once that is changed, I was thinking you can also rename `descend_array` to `serialize_array` to match the pattern, but now that I look at it, there's no longer a reason to have a separate method, and it can probably be inlined directly into this method.